include/aspect/freesurface.h

```diff
@@ -0,0 +1,246 @@
+       */
+      void setup_dofs();
+
+      /**
+       * Apply stabilization to a cell of the system matrix.  The
+       * stabilization is only added to cells on a free surface.  The
+       * scheme is based on that of Kaus et. al., 2010.  Called during
+       * assemly of the system matrix.
+       */
+      void apply_stabilization (const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                internal::Assembly::CopyData::StokesSystem<dim>      &data);
+
+      /**
+       * If a geometry model uses manifolds for the refinement behavior
+       * it can produce nonsensical meshes on refinement when using a
+       * free surface, since the free surface may deform the mesh to the
+       * point where the manifold is no longer a good description of the
+       * domain.  However, it can still be useful to have the manifold
+       * description for producing a nice starting mesh with the initial
+       * refinements (that is to say, before timestepping).  This detaches
+       * manifolds from cells, and is called after the initial refinements
+       * redistribution of the system.
+       * redistribution of the system.
+       * of the domain.
+       */
+      void detach_manifolds();
+
+      /**
+       * Declare parameters for the free surface handling.
+       */
+      static
+      void declare_parameters (ParameterHandler &prm);
+
+      /**
+       * Parse parameters for the free surface handling.
+       */
+      void parse_parameters (ParameterHandler &prm);
+
+    private:
+      /**
+       * Set the boundary conditions for the solution of the elliptic
+       * problem, which computes the displacements of the internal
+       * vertices so that the mesh does not become too distored due to
+       * motion of the free surface.  Velocities of vertices on the free
+       * surface are set to be the normal of the Stokes velocity solution
+       * projected onto that surface.  Velocities of vertices on free-slip
+       * boundaries are constrained to be tangential to those boundaries.
+       * Velocities of vertices on no-slip boundaries are set to be zero.
+       */
+      void make_constraints ();
+
+      /**
+       * Project the the Stokes velocity solution onto the
+       * free surface. Called by make_constraints()
+       */
+      void project_velocity_onto_boundary (LinearAlgebra::Vector &output);
+
+      /**
+       * Solve vector Laplacian equation for internal mesh displacements.
+       */
+      void compute_mesh_displacements ();
+
+      /**
+       * Calculate the velocity of the mesh for ALE corrections.
+       */
+      void interpolate_mesh_velocity ();
+
+      /**
+       * Reference to the Simulator object to which a FreeSurfaceHandler
+       * instance belongs.
+       */
+      Simulator<dim> &sim;
+
+      /**
+      * Finite element for the free surface implementation, which is
+      used for tracking mesh deformation.
```

Comment:
can you fix the indentation and lack of asterisk at the beginning of the second line?